db/index.js

```diff
@@ -0,0 +1,434 @@
+const { Pool } = require('pg');
+const readline = require('readline');
+// pools will use environment variables
+// for connection information
+const pool = new Pool({
+  database: 'productsapi',
+});
+const fs = require('fs');
+
+// Skus CSV Handler
+
+const skusFilename = './server/skus.csv';
+
+const skusReader = fs.createReadStream(skusFilename);
+const skusRl = readline.createInterface({
+  input: skusReader,
+  crlfdelay: Infinity,
+});
+skusRl.on('line', (line) => {
+  handleSkusLine(line);
+});
+let querySkusString = '';
+let countSkus = 0;
+
+let handleSkusLine = (line) => {
+  line = line.split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
+  for (let j = 0; j < line.length; j += 1) {
+    if (line[j][0] === '"') {
+      line[j] = line[j].substr(1);
+    }
+    if (line[j][line[j].length - 1] === '"') {
+      line[j] = line[j].substr(0, line[j].length - 1);
+    }
+    line[j] = line[j].replaceAll("'", "''");
+  }
+
+  querySkusString += `('${line[0]}', '${line[1]}', '${line[2]}', '${line[3]}'), `;
+  countSkus += 1;
+  if (countSkus === 1000) {
+    querySkusString = querySkusString.substring(0, querySkusString.length - 2);
+
+    pool.query(
+      `INSERT INTO skus(sku_id, style_id, size, quantity) VALUES ${querySkusString}`,
+      (skusErr, skusResult) => {
+        if (skusErr) {
+          console.error('Error executing query', skusErr.stack);
+        }
+        return skusResult;
+      }
+    );
+    querySkusString = '';
+    countSkus = 0;
+  }
+};
+skusRl.on('close', () => {
+  const extraSkusQuery = querySkusString.substring(0, querySkusString.length - 2);
```

Comment:
I'm having difficulty understanding this line. Why are the last 2 characters not included?

Reply:
The way that I have set up my long query strings there are two extra characters at the end that I have to get rid of so i don't get syntax errors with my queries. This is just removing those.